test/new-e2e/tests/orchestrator/apply.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/DataDog/test-infra-definitions/components/datadog/apps/redis"
+	dogstatsdstandalone "github.com/DataDog/test-infra-definitions/components/datadog/dogstatsd-standalone"
+	fakeintakeComp "github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+	localKubernetes "github.com/DataDog/test-infra-definitions/components/kubernetes"
+	resAws "github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+)
+
+// Apply creates a kind cluster, deploys the datadog agent, and installs various workloads for testing
+func Apply(ctx *pulumi.Context) error {
+	awsEnv, kindCluster, kindKubeProvider, err := createCluster(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create kind cluster: %w", err)
+	}
+
+	agentDependency, err := deployAgent(ctx, awsEnv, kindCluster, kindKubeProvider)
+	if err != nil {
+		return fmt.Errorf("failed to deploy agent to cluster: %w", err)
+	}
+
+	// Deploy testing workload
+	if awsEnv.TestingWorkloadDeploy() {
+		if _, err := redis.K8sAppDefinition(*awsEnv.CommonEnvironment, kindKubeProvider, "workload-redis", agentDependency); err != nil {
+			return fmt.Errorf("failed to install redis: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func createCluster(ctx *pulumi.Context) (*resAws.Environment, *localKubernetes.Cluster, *kubernetes.Provider, error) {
+	awsEnv, err := resAws.NewEnvironment(ctx)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	vm, err := ec2.NewVM(awsEnv, "kind")
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if err := vm.Export(ctx, nil); err != nil {
+		return nil, nil, nil, err
+	}
+
+	kindCluster, err := localKubernetes.NewKindCluster(*awsEnv.CommonEnvironment, vm, awsEnv.CommonNamer.ResourceName("kind"), awsEnv.KubernetesVersion())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if err := kindCluster.Export(ctx, nil); err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Export cluster’s properties
+	ctx.Export("kubeconfig", kindCluster.KubeConfig)
+
+	// Building Kubernetes provider
+	kindKubeProvider, err := kubernetes.NewProvider(ctx, awsEnv.Namer.ResourceName("k8s-provider"), &kubernetes.ProviderArgs{
+		EnableServerSideApply: pulumi.BoolPtr(true),
+		Kubeconfig:            kindCluster.KubeConfig,
+	})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return &awsEnv, kindCluster, kindKubeProvider, nil
+}
+
+const agentCustomValuesFmt = `
+datadog:
+  kubelet:
+    tlsVerify: false
+  clusterName: "%s"
+  orchestratorExplorer:
+    customResources:
+    - datadoghq.com/v1alpha1/datadogmetrics
+agents:
+  useHostNetwork: true
+
+clusterAgent:
+  enabled: true
+  confd:
+    orchestrator.yaml: |-
+      init_config:
+      instances:
+        - collectors:
+          - pods
+          - nodes
+          - deployments
+          - customresourcedefinitions
+          crd_collectors:
+          - datadoghq.com/v1alpha1/datadogmetrics
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I prefer embedded a yaml fil, to have yaml formatting support. But it is opinionated, cons is you don't see the content of the config

Reply:
I can get behind that - it grew from the small file I based the test on

88c1e5cd525754e6d8733db28bcc78d56445c7a6